Source/WebCore/inspector/agents/InspectorPageAgent.cpp

```diff
@@ -284,6 +284,9 @@ InspectorPageAgent::ResourceType InspectorPageAgent::inspectorResourceType(Cache
 
 InspectorPageAgent::ResourceType InspectorPageAgent::inspectorResourceType(const CachedResource& cachedResource)
 {
+    if (cachedResource.type() == CachedResource::Type::MainResource && MIMETypeRegistry::isSupportedImageMIMEType(cachedResource.mimeType()))
```

Comment:
are there any other things we could/should do like this?

e.g. if you're directly loading a `.js` resource should we have it be a `InspectorPageAgent::ScriptResource`?

Reply:
Razvan and I chatted about this this morning when discussing this approach - it would likely be technically correct to do so, but it is also less important to do so since the fallback is to show text form of content anyways. This specific fix is targeting a known bug - a follow up to expand upon this fix wouldn’t hurt though.